### PR TITLE
Create and update user endpoints

### PR DESCRIPTION
**Pull request by 04HuynhM**
Title.

### CODE REVIEW

File: functions/index.js [hunk 1]
```diff
@@ -5,13 +5,15 @@ admin.initializeApp(functions.config().firebase);
...
+const users = require('./users');
 
 const funs = {
   helloWorld: functions.https.onRequest((req, res) => {
     res.send('Hello from a Serverless Database!');
   }),
   ...rooms,
-  ...devices
+  ...devices,
+  ...users
```
Comment (thefnox) [inline functions/index.js]: You're about to haver a merge conflict with the other pr

File: functions/users/index.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    if(req.method !== 'POST') {
+        return res.status(400).json({
+            message: 'Bad request, POST required'
+        })
+    }
+    else {
+        const data = req.body;
+        if (!checkUserData(data)) {
+            return res.status(400).json({
+                message: 'Missing parameters ' +
```
Comment (thefnox) [inline functions/users/index.js]: There's no need for concatenation here, this is a single constant

File: functions/users/index.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            })
+        }
+        admin.auth().createUser({
+            email: data.email,
+            emailVerified: false,
+            password: data.password,
+            disabled: false
+        })
+            .then((userRecord) => {
+                console.log("User created with UID of: " + userRecord.uid);
```
Comment (thefnox) [inline functions/users/index.js]: Prefer template strings to concatenation, 
```suggestion
                console.log(`User created with UID of: ${userRecord.uid}`);
```

File: functions/users/index.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+exports.updateUser = functions.https.onRequest((req, res) => {
+    if(req.method !== 'POST') {
+        return res.status(400).json({
+            message: "Bad request, POST Required"
+        })
+    } else {
+        const data = req.body;
+        dbref.child('Users').child(data.uid).update({
+            [data.updatedType] : data.value.toString()
```
Comment (thefnox) [inline functions/users/index.js]: Yikes, this allows the user to modify all attributes? Shouldn't we restrict to specific ones?

File: functions/users/index.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+                }).then(() => {
+                    return res.status(200).json({
+                        message: 'User Created Successfully',
+                        email: userRecord.email,
+                        user_uid: userRecord.uid,
+                    });
+                })
+            })
+            .catch((error) => {
+                console.log("Error creating user: ", error);
```
Comment (thefnox) [inline functions/users/index.js]: Same here

**Review by thefnox — CHANGES_REQUESTED** [review]

File: functions/users/index.js (hunk 4 above)
Comment (04HuynhM) [reply functions/users/index.js]: Yeah, good point. We only need to restrict email though.

**Pull request closed without merging**